src/System.Private.CoreLib/src/System/Resources/ResourceManager.cs

```diff
@@ -1069,9 +1061,19 @@ internal UltimateResourceFallbackLocation FallbackLoc
                 set { _rm._fallbackLoc = value; }
             }
 
-            internal RuntimeAssembly MainAssembly
+            internal Assembly MainAssembly
+            {
+                get { return _rm.MainAssembly; }
+            }
+
+            internal string MainAssemblySimpleName
             {
-                get { return (RuntimeAssembly)_rm.MainAssembly; }
+                get
+                {
+                    if (_rm.MainAssembly is RuntimeAssembly runtimeAssembly)
```

Comment:
This special casing for RuntimeAssembly does not hurt, but it should not be necessary. This property is only used for exception handling where performance does not matter a lot.

Reply:
Ok, removed it. Makes it easier to read the code.